docs/development/design/telemetry.rst

```diff
@@ -0,0 +1,232 @@
+         "name": "alabaster",
+         "platform": "noarch",
+         "version": "0.7.12"
+      },
+      {
+         "base_url": "https://conda.anaconda.org/conda-forge",
+         "build_number": 0,
+         "build_string": "pyh9f0ad1d_0",
+         "channel": "conda-forge",
+         "dist_name": "asn1crypto-1.4.0-pyh9f0ad1d_0",
+         "name": "asn1crypto",
+         "platform": "noarch",
+         "version": "1.4.0"
+      },
+      {
+         "base_url": "https://conda.anaconda.org/conda-forge",
+         "build_number": 3,
+         "build_string": "3",
+         "channel": "conda-forge",
+         "dist_name": "python-3.5.4-3",
+         "name": "python",
+         "platform": "linux-64",
+         "version": "3.5.4"
+      }
+   ]
+
+APT packages
+~~~~~~~~~~~~
+
+This isn't implemented yet, but when it is,
+we can get the list from the config file,
+or we can list the packages installed with ``dpkg --get-selections``.
+That command would list all pre-installed packages as well, so we may be getting some noise.
+
+.. code-block::
+
+   $ dpkg --get-selections
+
+   adduser                                         install
+   apt                                             install
+   base-files                                      install
+   base-passwd                                     install
+   bash                                            install
+   binutils                                        install
+   binutils-common:amd64                           install
+   binutils-x86-64-linux-gnu                       install
+   bsdutils                                        install
+   build-essential                                 install
+
+Python
+~~~~~~
+
+We can get the Python version from the config file when using a Python environment,
+and from the ``conda list`` output when using a Conda environment.
+
+OS
+~~
+
+We can infer the OS version from the build image used in the config file,
+but since it changes with time, we can get it from the OS itself:
+
+.. code-block::
+
+   $ lsb_release --description
+   Description:    Ubuntu 18.04.5 LTS
+   # or
+   $ cat /etc/issue
+   Ubuntu 18.04.5 LTS \n \l
+
+Storage
+-------
+
+We can save all this information in json files in cloud storage,
+then we could use a tool to import all this data into.
+Or we can decide for a tool or service where to fed all this data directly into.
+
+If we decide to save the files in cloud storage,
+we can try to calculate a hash of the file so we don't upload duplicates that happen on the same day/month.
+We can aggregate this data per year/month saving them in following structure:
+``telemetry/builds/{year}/{month}/{year}-{month}-{day}-{timestamp-pk|pk}.json``,
+that way is easy to download, all data per year/month without iterating over all files.
+
+.. Since this information isn't sensitive,
+   I think we are fine with this structure
+   (we can't do bulk deletes of all info about a project if we follow this structure).
```

Comment:
I think storing it in "cold" storage is right for the long-term large-scale analysis. I'd love to have a subset of important things look more like our monitoring data, and reported/graphed live via cloudwatch. We could pick the top 10 most important/interesting things (eg. theme, sphinx version, etc.)